pallets/utxo/src/lib.rs

```diff
@@ -230,12 +230,53 @@ pub mod pallet {
         }
     }
 
+    #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+    #[derive(Clone, Encode, Decode, Eq, PartialEq, PartialOrd, Ord, RuntimeDebug, Hash)]
+    pub enum TxHeaderAndExtraData {
+        NormalTx {
+            // Normal token ID
+            id: TokenId,
+        },
+        ConfidentialTx, /* not implemented yet */
+        Nft {
+            id: TokenId,
+            data_hash: [u8; 32],
```

Comment:
Why are we storing the data hash with the header? This is kind of redundant. Shouldn't the transaction hash already have this information? If you need the data hash, you can just calculate it.

Reply:
It's a hash of remote digital data that not include in transactions. It's impossible to calculate it without sending in a transaction all digital data that is unnecessary and might be huge.

Reply:
As we told before I should it rename to `data` and remove restrictions about data size. I've done it.